pynest/examples/store_restore_network.py

```diff
@@ -0,0 +1,380 @@
+
+        self._next_line += 1
+
+
+if __name__ == "__main__":
+
+    plt.ion()
+
+    T_sim = 1000
+
+    dplot = DemoPlot()
+
+    ###############################################################################
+    # Ensure clean slate and make NEST less chatty
+    nest.set_verbosity("M_WARNING")
+    nest.ResetKernel()
+
+    ###############################################################################
+    # Create network from scratch and simulate 1s.
+    nest.SetKernelStatus({"local_num_threads": 4,
+                          "print_time": True})
+    ein = EINetwork()
+
+    print("*** Initial simulation ***")
+    ein.build()
+    nest.Simulate(T_sim)
+    dplot.add_to_plot(ein, lbl="Initial simuation")
+
+    ###############################################################################
+    # Store network state to file with state after 1s.
+    print("\n*** Storing simulation ...", end="", flush=True)
+    ein.store("ein_1000.pkl")
+    print(" done ***\n")
+
+    ###############################################################################
+    # Continue simulation by another 1s.
+    print("\n*** Continuing simulation ***")
+    nest.Simulate(T_sim)
+    dplot.add_to_plot(ein, lbl="Continued simuation", t_min=T_sim, t_max=2*T_sim)
+
+    ###############################################################################
+    # Clear kernel, restore network from file and simulate for 1s.
+    print("\n*** Reloading and resuming simulation ***")
+    nest.ResetKernel()
+    nest.SetKernelStatus({"local_num_threads": 4})
+    ein2 = EINetwork()
+    ein2.restore("ein_1000.pkl")
+    nest.Simulate(T_sim)
+    dplot.add_to_plot(ein2, lbl="Reloaded simuation")
+
+    ###############################################################################
+    # Repeat previous step. This shall result in *exactly* the same results as
+    # the previous run because we use the same random seed.
+    print("\n*** Reloading and resuming simulation (same seed) ***")
+    nest.ResetKernel()
+    nest.SetKernelStatus({"local_num_threads": 4})
+    ein2 = EINetwork()
+    ein2.restore("ein_1000.pkl")
+    nest.Simulate(T_sim)
+    dplot.add_to_plot(ein2, lbl="Reloaded simuation (same seed)")
+
+    ###############################################################################
+    # Clear, restore and simulate again, but now with different random seed.
+    # Details in results shall differ from previous run.
+    print("\n*** Reloading and resuming simulation (different seed) ***")
+    nest.ResetKernel()
+    nest.SetKernelStatus({"local_num_threads": 4, "rng_seed": 987654321})
+    ein2 = EINetwork()
+    ein2.restore("ein_1000.pkl")
+    nest.Simulate(T_sim)
+    dplot.add_to_plot(ein2, lbl="Reloaded simulation (different seed)")
+
+    dplot.fig.savefig("store_restore_network.png")
+
+    input("Press ENTER to close figure!")
+
+    # NOTE: Comments below need updating after discussion about model to be used.
```

Comment:
Remove this comment block as information is already in the figure legend?

Reply:
Done